Amplify/Categories/DataStore/Model/DataStoreSchema.swift

```diff
@@ -9,7 +9,7 @@ import Foundation
 
 /// Protocol that defines a contract between the consumer and the DataStore plugin.
 /// All models have to be registered and have an associated `version`.
-public protocol AmplifyModelRegistration {
+public protocol DataStoreSchema {
```

Comment:
How about `SchemaRegistry` or `ModelSchema`? Having "DataStore" in APIPlugin seems odd

Reply:
You guys probably already realized this, but if we make this change, we'll need to update the codegen tools so that it generates:
```
final public class AmplifyModels: DataStoreSchema {
```
instead of what it currently geneates:
```
final public class AmplifyModels: AmplifyModelRegistration {
```

Reply:
> How about `SchemaRegistry` or `ModelSchema`? Having "DataStore" in APIPlugin seems odd

I understand the concern, but we need to keep in mind that the model-based API integration should be revisited. Right now API imports classes and protocols from the DataStore category, so the integration is there already. We need to think of ways of making that more isolated.

That said, `ModelSchema` cannot be used because it's already used by `Model`, as the schema of each `Model`. That said, I'll think of a better name for it, `SchemaRegistry` could be one.

Reply:
> Comments about aligning naming --
> 
> For your consideration, on Android, we have two similar components which are named.
> 
> 1. The [`ModelProvider`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/core/model/ModelProvider.java#L20-L24), and its code-generated `AmplifyModelProvider` ([an example of one](https://github.com/aws-amplify/amplify-android/blob/master/testmodels/src/main/java/com/amplifyframework/testmodels/commentsblog/AmplifyModelProvider.java#L10-L15)). This component is read-only and declares what the different types of models are, and a version for their contract, as a whole.
> 2. The [`ModelSchemaRegistry`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/core/model/ModelSchemaRegistry.java#L27-L30), which accepts a collection of model classes as inputs, and then constructs and maintains [`ModelSchema`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/core/model/ModelSchema.java#L44-L48) for each, by parsing Java annotations and using Java reflection, on the `@Model`-annotated classes.
> 
> To recall, as above, Android generates the schema at runtime, not compile-time. So, there are some architectural differences. But we can probably keep the names a little tighter between the platforms even so, if we try.

thanks @jamesonwilliams for the call-out. JavaScript simply calls it `schema` (using `initSchema(schema: Schema)`. Before iOS was using `ModelRegistration` which is odd, so this is an attempt to align iOS more with JS. I'll revisit the Android and JS implementations to see if we can agree on something. I'll get in touch with you directly before I submit an update to this PR.

Reply:
@jamesonwilliams @wooj2 @lawmicha I updated to `ModelSchemaProvider`. Jameson, do you think it's OK to rename `ModelProvider` to `ModelSchemaProvider` on your end?

Reply:
Sorry, I may have accidentally just created more confusion. :-(

On Android, the CLI code-gens an `AmplifyModelProvider implements ModelProvider`; the `ModelProvider` interface returns a `Set<Class<? extends Model>>`.

At runtime, we access that `Set<Class<? extends Model>>`. For each `modelClass` in that set, we establish a 1:1 relationship to a `ModelSchema`.  We build the `ModelSchema`s in the library at run-time, by inspecting each `Class<? extends Model>` instance via reflection. We end up with a `List<ModelSchema>`, that is held in-memory in a 
 class called `ModelSchemaRegistry`.